app/services/converters/order_detail_to_journal_rows.rb

```diff
@@ -0,0 +1,23 @@
+module Converters
+  class OrderDetailToJournalRows
+
+    attr_accessor :journal, :order_detail, :total
+
+    def initialize(journal, order_detail, options = {})
+      @journal = journal
+      @order_detail = order_detail
+      @total = options[:total] || order_detail.total
+    end
+
+    def to_a
+      [{
+        account: order_detail.product.account,
```

Comment:
I was attempting to port the existing logic (to adhere the single responsibility principle) without altering the existing logic too much.  Seems like it will be safer to take baby steps.